extensions/interactions/TextInput/directives/text-input-validation.service.spec.ts

```diff
@@ -22,43 +22,49 @@ import { AnswerGroup, AnswerGroupObjectFactory } from
   'domain/exploration/AnswerGroupObjectFactory';
 import { AppConstants } from 'app.constants';
 import { InteractionSpecsConstants } from 'pages/interaction-specs.constants';
-import { OutcomeObjectFactory } from
-  'domain/exploration/OutcomeObjectFactory';
-import { Rule, RuleObjectFactory } from
-  'domain/exploration/RuleObjectFactory';
-import { SubtitledUnicode } from
-  'domain/exploration/SubtitledUnicodeObjectFactory';
-import { TextInputValidationService } from
-  'interactions/TextInput/directives/text-input-validation.service';
+import { Outcome, OutcomeObjectFactory } from 'domain/exploration/OutcomeObjectFactory';
+import { Rule, RuleObjectFactory } from 'domain/exploration/RuleObjectFactory';
+import { SubtitledUnicode } from 'domain/exploration/SubtitledUnicodeObjectFactory';
+import { TextInputValidationService } from 'interactions/TextInput/directives/text-input-validation.service';
+import { TextInputCustomizationArgs } from 'interactions/customization-args-defs';
 
 describe('TextInputValidationService', () => {
-  let validatorService, WARNING_TYPES;
-  let INTERACTION_SPECS, customizationArgSpecs, rowsSpecs, minRows, maxRows;
-
-  let currentState, customizationArguments;
-  let goodAnswerGroups, goodDefaultOutcome;
-  let oof, agof, rof;
+  let validatorService: TextInputValidationService;
+  let WARNING_TYPES = AppConstants.WARNING_TYPES;
+  let INTERACTION_SPECS = InteractionSpecsConstants.INTERACTION_SPECS;
+  let minRows: number | undefined;
+  let maxRows: number | undefined;
+
+  let currentState: string;
+  let customizationArguments: TextInputCustomizationArgs;
+  let goodAnswerGroups: AnswerGroup[];
+  let goodDefaultOutcome: Outcome;
+  let oof: OutcomeObjectFactory;
+  let agof: AnswerGroupObjectFactory;
+  let rof: RuleObjectFactory;
 
   let createAnswerGroupByRules: (rules: Rule[]) => AnswerGroup;
 
   beforeEach(() => {
-    validatorService = TestBed.get(TextInputValidationService);
-    oof = TestBed.get(OutcomeObjectFactory);
-    agof = TestBed.get(AnswerGroupObjectFactory);
-    rof = TestBed.get(RuleObjectFactory);
+    validatorService = TestBed.inject(TextInputValidationService);
+    oof = TestBed.inject(OutcomeObjectFactory);
+    agof = TestBed.inject(AnswerGroupObjectFactory);
+    rof = TestBed.inject(RuleObjectFactory);
     WARNING_TYPES = AppConstants.WARNING_TYPES;
-    INTERACTION_SPECS = InteractionSpecsConstants.INTERACTION_SPECS;
-    customizationArgSpecs = INTERACTION_SPECS.TextInput.customization_arg_specs;
-    rowsSpecs = customizationArgSpecs[1];
-    minRows = rowsSpecs.schema.validators[0].min_value;
-    maxRows = rowsSpecs.schema.validators[1].max_value;
+    let customizationArgSpecs =
+     INTERACTION_SPECS.TextInput.customization_arg_specs;
+    let rowsSpecs = customizationArgSpecs[1];
+    if (rowsSpecs.schema.validators) {
```

Comment:
please update this by type casting

Reply:
Not sure if we could typecast it. This value comes from a json file. Can you take a look ?

Reply:
try something like this
```typescript
    const validators = <({
        'min_value': number;
        id: string;
        'max_value'?: undefined;
    } | {
        'max_value': number;
        id: string;
        'min_value'?: undefined;
    })[]> rowsSpecs.schema.validators;
    minRows = validators[0].min_value;
    maxRows = validators[1].max_value;
```

Reply:
done